core/shared/src/main/scala/zio/prelude/These.scala

```diff
@@ -32,6 +32,13 @@ sealed trait These[+A, +B] { self =>
       case (Both(l1, r1), Both(l2, r2)) => Both(A1.combine(l1, l2), B1.combine(r1, r2))
     }
 
+  def fold[C](left: A => C, right: B => C, both: (A, B) => C): C =
```

Comment:
I think I would curry the `both` parameter since that is a function from two arguments two one while the others are functions from one argument to one though that is slightly subjective. That is also consistent with methods like the signature of operators like `ZStream.zipAllWith`.

Reply:
Thanks for the review.
Will do